### ISSUE TITLE
"Ignored an invalid @Startup method" with Kafka-Streams

### ISSUE THREAD

**Issue by wicksim** [quarkusio/quarkus#36774]
### Describe the bug

When starting Quarkus with kafka-streams-extension enabled, the following warnings are printed in the log:

`[WARNING] [io.quarkus.arc.deployment.StartupBuildSteps] Ignored an invalid @Startup method declared on io.quarkus.kafka.streams.runtime.KafkaStreamsProducer: org.apache.kafka.streams.KafkaStreams getKafkaStreams()
[WARNING] [io.quarkus.arc.deployment.StartupBuildSteps] Ignored an invalid @Startup method declared on io.quarkus.kafka.streams.runtime.KafkaStreamsProducer: io.quarkus.kafka.streams.runtime.KafkaStreamsTopologyManager kafkaStreamsTopologyManager()`

### Expected behavior

No warnings should be printed.

### Actual behavior

Warnings are printed.

### How to Reproduce?

1. Create Quickstart-project with kafka-streams-extention
2. `mvnw clean install`
3. See output

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19045.3570]

### Output of `java -version`

OpenJDK Runtime Environment Corretto-17.0.9.8.1 (build 17.0.9+8-LTS)

### Quarkus version or git rev

3.5.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.5 (57804ffe001d7215b5e7bcb531cf83df38f93546)

**Comment by ozangunalp** [comment]
I've seen this in the past. I'll have a look

**Comment by alesj** [comment]
It's the @Produces that's the culprit ...

```
                    if (annotationStore.hasAnnotation(method, STARTUP_NAME)) {
                        if (!method.isSynthetic()
                                && !Modifier.isPrivate(method.flags())
                                && !Modifier.isStatic(method.flags())
                                && method.parametersCount() == 0
                                && !annotationStore.hasAnnotation(method, DotNames.PRODUCES)) {
                            startupMethods.add(method);
                        } else {
                            LOG.warnf("Ignored an invalid @Startup method declared on %s: %s", method.declaringClass().name(),
                                    method);
                        }
                    }
```

**Comment by mkouba** [comment]
> It's the @produces that's the culprit ...
> 
> ```
>                     if (annotationStore.hasAnnotation(method, STARTUP_NAME)) {
>                         if (!method.isSynthetic()
>                                 && !Modifier.isPrivate(method.flags())
>                                 && !Modifier.isStatic(method.flags())
>                                 && method.parametersCount() == 0
>                                 && !annotationStore.hasAnnotation(method, DotNames.PRODUCES)) {
>                             startupMethods.add(method);
>                         } else {
>                             LOG.warnf("Ignored an invalid @Startup method declared on %s: %s", method.declaringClass().name(),
>                                     method);
>                         }
>                     }
> ```

Yes, the [documentation](https://quarkus.io/guides/lifecycle#startup_annotation) is pretty clear that _"The `@Startup` annotation can be also declared on a non-static non-producer no-args method"_.

If you need to initialize the beans backed by the producer methods during startup you would need something like this instead:
```java
void onStart(@Observes StartupEvent event, KafkaStreams ks, KafkaStreamsTopologyManager kstm) {}
```
The injection points trigger the producers.

**Comment by lukens** [comment]
I'm getting these warnings too. I think from the other comments that this is a Quarkus thing, and not a me thing. Is that right?

**Comment by mkouba** [comment]
> I'm getting these warnings too. I think from the other comments that this is a Quarkus thing, and not a me thing. Is that right?

Yes. Specifically, the `quarkus-kafka-streams` extension declares invalid `@Startup` methods here: https://github.com/quarkusio/quarkus/blob/main/extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsProducer.java#L119-L133. As I explained in the comment above - a regular `StartupEvent` observer is needed instead.

CC @ozangunalp @alesj

**Comment by ozangunalp** [comment]
Fixed by #38705
